File: 2021/september/day-28/solution1.py
class Solution:
    def sortArrayByParityII(self, nums: List[int]) -> List[int]:
        N = len(nums)
        i,j = 0,1
        while i<N and j<N:
            if nums[i]%2==0:
                i+=2
            elif nums[j]%2!=0:
                j+=2
            else:
                nums[i],nums[j] = nums[j],nums[i]
                i+=2
                j+=2
        
        return nums

File: 2021/october/day-6/solution1.cpp
class Solution {
    public:
        vector<int> findDuplicates(vector<int>& nums) {
            //we can't use map or set as they take more space
            // one idea is to sort and check adjacent elements ut in that case time
            // complexity will be O(nlogn)
            //now it's given that numbers in array ae less than equal to length of array and all are positive
            // the idea is to traverse the array and go at the position of the number encounterd and make them negative, if they are already negative store abs value of n in another array as it will be going to the same position 2nd time.
            
            vector<int> results;
            for(int n:nums){
               
                if(nums[abs(n)-1] > 0) nums[abs(n)-1] *=-1;
                else
                    results.push_back(abs(n));
                
            }
            return results;
            
        }
    };